dot/services.go

```diff
@@ -387,6 +387,7 @@ func createSyncService(cfg *Config, st *state.Service, bp sync.BlockProducer, fg
 		Verifier:         verifier,
 		Runtime:          rt,
 		DigestHandler:    dh,
+		BaseState:        st.Base,
```

Comment:
it might be nicer to pass in the code substitutes here instead of the BaseState, so the sync service doesn't need to deal with the conversions

Reply:
Updated.